react/modules/tasks/nextjs-ssr-ssg.md

```diff
@@ -0,0 +1,41 @@
+# Next.js. Server Side Rendering
+
+## Technical requirements
+
+1. Create a separate branch for this task from the previous branch task. Branch name: "nextjs-ssr-pages-api".
+2. Next.js Pages API Integration:
+   - Migrate your application from vite to next.js by using Pages folder API. You'll need to adapt existing libraries to work with next.js.
+3. Next.js App Router API Integration:
+   - Create a separate branch for this point from "nextjs-ssr-pages-api" branch. Branch name: "nextjs-ssr-app-router-api".
+   - Migrate from Pages Api to App Folder Api.
```

Comment:
For this 2nd task, could they use the initial setup (vite) and just migrate it right to the App Folder api? So they could work on 2 NextJS tasks in parallel?

Reply:
I don't think so. There is helpful guide to do this task in right way https://nextjs.org/docs/pages/building-your-application/upgrading/app-router-migration.  It also can be usefull for future job if stuidents will work with legacy next.js projects.